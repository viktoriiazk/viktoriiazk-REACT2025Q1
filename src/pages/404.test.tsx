import { render, screen } from '@testing-library/react';
import Page404 from './404.tsx';

describe('Page404', () => {
  it('renders the 404 message', () => {
    render(<Page404 />);
    expect(screen.getByText('404 - Page Not Found')).toBeInTheDocument();
  });

  it('renders a heading with "404 - Page Not Found"', () => {
    render(<Page404 />);
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toHaveTextContent('404 - Page Not Found');
  });

  it('has the correct class applied to the div element', () => {
    render(<Page404 />);
    const divElement = screen.getByTestId('error-page');
    expect(divElement).toHaveClass('error-page');
  });
});
