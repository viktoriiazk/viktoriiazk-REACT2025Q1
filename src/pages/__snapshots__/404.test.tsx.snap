// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page404 > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      404 - Page Not Found
    </h1>
  </div>
</DocumentFragment>
`;
